doc/reference/kernel/scheduling/index.rst

```diff
@@ -31,6 +31,74 @@ exist, the scheduler chooses the one that has been waiting longest.
     at any time unless interrupts have been masked. This applies to both
     cooperative threads and preemptive threads.
 
+
+The kernel can be built with with several choices for the
+ready queue implementation, offering different choices between
+code size, constant factor runtime overhead and performance
+scaling when many threads are added.
+
+* Simple linked-list ready queue
+
+  The scheduler ready queue will be implemented as a simple unordered list, with
+  very fast constant time performance for single threads and very low code size.
+  This implementation should be selected on systems with constrained code size
+  that will never see more than a small number (3, maybe) of runnable threads in
+  the queue at any given time.  On most platforms (that are not otherwise using
+  the red/black tree) this results in a savings of ~2k of code size.
+
+* Red/black tree ready queue"
+
+  The scheduler ready queue will be implemented as a red/black tree.  This has
+  rather slower constant-time insertion and removal overhead, and on most
+  platforms (that are not otherwise using the rbtree somewhere) requires an
+  extra ~2kb of code.  But the resulting behavior will scale cleanly and
+  quickly into the many thousands of threads.
+
+  Use this on platforms where you may have many threads (very roughly: more
+  than 20 or so) marked as runnable at a given time.
+
+  .. note::
+
+      Most applications don't want this.
```

Comment:
we don't really want to use the ``.. note`` directive for casual aside comments, so better to just add the sentence to the end of the previous paragraph:
```
Use this for applications needing many concurrent runnable threads (> 20 or so).
Most applications won't need this ready queue implementation.
```